File: conformance/bundles/14-stateless-steps/squares.steps.rs
//! Rust sibling of `squares.steps.ts` (bundle `14-stateless-steps`).
//!
//! Pure steps — nothing to arrange or evolve — so the context is the unit type.

use varar::Steps;

pub fn register(s: &mut Steps<()>) {
    s.stimulus("I warm up my mental math", |ctx: ()| Ok(ctx));
    // Two slots ({int}, {int}): two ints in, two ints out, compared
    // positionally — the same contract as the .ts sibling's `[n, n * n]`.
    s.sensor("The square of {int} is {int}.", |_ctx: (), n: i64, _square: i64| Ok((n, n * n)));
}

pub fn state() {}

File: conformance/bundles/14-stateless-steps/squares.steps.go
// Go sibling of squares.steps.ts (bundle 14-stateless-steps).
//
// Pure steps — nothing to arrange or evolve — so State() is the bare null every
// handler ignores. Written in the typed form: each {int} arrives as an int, and
// the two returned ints are compared positionally against the two slots, the
// same contract as the .ts sibling's `[n, n * n]` tuple.
package fixture

import "github.com/varar-dev/varar-go/varar"

func Register(s *varar.Steps) {
	s.Stimulus("I warm up my mental math", func(state varar.Value) (varar.Value, error) {
		return state, nil
	})
	s.Sensor("The square of {int} is {int}.",
		func(state varar.Value, n, square int) (int, int, error) {
			return n, n * n, nil
		})
}

func State() varar.Value {
	return varar.NullValue
}
